python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,379 @@
+        # such as: "Field with name 'coord_raErr' not found with type 'F'").
+        afwTable.CoordKey.addErrorFields(self.schema)
+
+        self.subtasks = ["detection", "deblend", "measurement"]
+
+    def make_subtasks(self):
+        """Create subtasks based on the current configuration."""
+        for name in self.subtasks:
+            self.makeSubtask(name, schema=self.schema)
+
+    def run(
+        self,
+        image,
+        bands=None,
+        band=None,
+        mask=None,
+        variance=None,
+        psf=None,
+        wcs=None,
+        photo_calib=None,
+        id_generator=None,
+    ):
+        """Run detection, optional deblending, and measurement on a given
+        image.
+
+        Parameters
+        ----------
+        image: `~lsst.afw.image.Exposure` or `~lsst.afw.image.MaskedImage` or
+            `~lsst.afw.image.Image` or `np.ndarray` or
+            `~lsst.afw.image.MultibandExposure` or
+            `list` of `~lsst.afw.image.Exposure`
+            The image on which to detect, deblend and measure sources. If
+            provided as a multiband exposure, or a list of `Exposure` objects,
+            it can be taken advantage of by the 'scarlet' deblender. When using
+            a list of `Exposure` objects, the ``bands`` parameter must also be
+            provided.
+        bands: `str` or `list` of `str`, optional
+            The bands of the input image. Required if ``image`` is provided as
+            a list of `Exposure` objects. Example: ["g", "r", "i", "z", "y"]
+            or "grizy".
+        band: `str`, optional
+            The target band of the image to use for detection and measurement.
+            Required when ``image`` is provided as a `MultibandExposure`, or a
+            list of `Exposure` objects.
+        mask: `~lsst.afw.image.Mask`, optional
+            The mask for the input image. Only used if ``image`` is provided
+            as an afw `Image` or a numpy `ndarray`.
+        variance: `~lsst.afw.image.Image`, optional
+            The variance image for the input image. Only used if ``image`` is
+            provided as an afw `Image` or a numpy `ndarray`.
+        psf: `~lsst.afw.detection.Psf`, optional
+            The PSF model for the input image. Will be ignored if ``image`` is
+            provided as an `Exposure`, `MultibandExposure`, or a list of
+            `Exposure` objects.
+        wcs: `~lsst.afw.image.Wcs`, optional
+            The World Coordinate System (WCS) model for the input image. Will
+            be ignored if ``image`` is provided as an `Exposure`,
+            `MultibandExposure`, or a list of `Exposure` objects.
+        photo_calib : `~lsst.afw.image.PhotoCalib`, optional
+            Photometric calibration model for the input image. Will be ignored
+            if ``image`` is provided as an `Exposure`, `MultibandExposure`, or
+            a list of `Exposure` objects.
```

Comment:
I think that this API is confusing. Instead of having three optional sets of parameters in `run`, one for `Exposure`s, and one for `Image`s (which requires a lot of additional parameters that are ignored if the `image` is an `Exposure`), and one for `MultibandExposure`s, I recommend making run take an `Exposure` and create a new method, `runFromImage` that takes the parameters required if an image is passed and creates an exposure from those images to call the normal `run`. I think that this will make the API easier for users and simplify the codebase. See my other comments about single-band vs multiband

Reply:
I like all these suggestions. I implemented them in the new version.